### ISSUE TITLE
Generated encoder for a case class with None produces invalid json

### ISSUE THREAD

**Issue by alvaroc1** [zio/zio-json#91]
```scala
case class User (name: Option[String], age: Int)
object User { 
  implicit val encoder = DeriveJsonEncoder.gen[User]
}
```
```
scala> User(None, 37).toJson
res7: String = {,"age":37}
```

Note the extra comma at the beginning.

If the fields are re-ordered, the json is fine:
```scala
case class User (age: Int, name: Option[String])
object User { 
  implicit val encoder = DeriveJsonEncoder.gen[User]
}
```
```
scala> User(36, None).toJson
res9: String = {"age":36}
```

**Comment by fsvehla** [comment]
Thanks for the report, @alvaroc1.

**Comment by alvaroc1** [comment]
I'll take a crack at fixing it

**Comment by fsvehla** [comment]
Closed by #92
